crates/bevy_feathers/src/theme.rs

```diff
@@ -0,0 +1,308 @@
+use bevy_app::Propagate;
+use bevy_color::{palettes, Color};
+use bevy_ecs::{
+    change_detection::DetectChanges,
+    component::Component,
+    lifecycle::Insert,
+    observer::On,
+    query::Changed,
+    resource::Resource,
+    system::{Commands, Query, Res},
+};
+use bevy_log::warn_once;
+use bevy_platform::collections::HashMap;
+use bevy_text::TextColor;
+use bevy_ui::{BackgroundColor, BorderColor};
+
+/// A collection of properties that make up a theme.
+#[derive(Default, Clone)]
+pub struct ThemeProps {
+    /// Map of design tokens to colors.
+    pub color: HashMap<String, Color>,
+    // Other style property types to be added later.
+}
+
+/// The currently selected user interface theme. Overwriting this resource changes the theme.
+#[derive(Resource, Default)]
+pub struct UiTheme(pub ThemeProps);
+
+impl UiTheme {
+    /// Lookup a color by design token. If the theme does not have an entry for that token,
+    /// logs a warning and returns an error color.
+    pub fn color<'a>(&self, token: &'a str) -> Color {
+        let color = self.0.color.get(token);
+        match color {
+            Some(c) => *c,
+            None => {
+                warn_once!("Theme color {} not found.", token);
+                // Return a bright obnoxious color to make the error obvious.
+                palettes::basic::FUCHSIA.into()
+            }
+        }
+    }
+
+    /// Associate a design token with a given color.
+    pub fn set_color(&mut self, token: impl Into<String>, color: Color) {
+        self.0.color.insert(token.into(), color);
+    }
+}
+
+/// Component which causes the background color of an entity to be set based on a theme color.
+#[derive(Component, Clone, Copy)]
+#[require(BackgroundColor)]
+#[component(immutable)]
+pub struct ThemeBackgroundColor(pub &'static str);
+
+/// Component which causes the border color of an entity to be set based on a theme color.
+/// Only supports setting all borders to the same color.
+#[derive(Component, Clone, Copy)]
+#[require(BorderColor)]
+#[component(immutable)]
+pub struct ThemeBorderColor(pub &'static str);
+
+/// Component which causes the inherited text color of an entity to be set based on a theme color.
+#[derive(Component, Clone, Copy)]
+#[component(immutable)]
+pub struct ThemeFontColor(pub &'static str);
+
+/// A marker component that is used to indicate that the text entity wants to opt-in to using
```

Comment:
The naming feels at odds with the very narrow scope. I'd either rename this or tweak the docs to cover the intended scope, then say "currently this only does inherited text styling".